src/fromager/sources.py

```diff
@@ -311,17 +309,21 @@ def patch_source(
 
 
 def _warn_for_old_patch(
-    source_root_dir: pathlib.Path, patch_filename: pathlib.Path, version: Version
+    patches_dir: pathlib.Path, source_root_dir: pathlib.Path
 ) -> None:
-    # Get the existing version and req name as per the source_root_dir naming conventions
+    # Get the req name as per the source_root_dir naming conventions
     req_name = source_root_dir.name.rsplit("-", 1)[0]
-    existing_version = Version(source_root_dir.name.rsplit("-", 1)[1])
 
-    # Check to see if patches for older version exists
-    if patch_filename.parent.name == source_root_dir.name:
-        logger.warning(
-            f"{req_name}: patches for version {existing_version!s} exist but will not be applied"
-        )
+    # Get all the patches for that requirement
+    req_patches = overrides.patches_for_req(patches_dir, req_name)
+
+    # Check if old patches exist and current patch not in the list
+    if len(req_patches) > 0:
+        string_to_search = "\t".join(str(p) for p in req_patches)
+        if source_root_dir.name not in string_to_search:
```

Comment:
I'm worried that relying on substrings is going to be brittle because of the name collisions I mentioned in the earlier comment.

If you build 2 sets, one containing all patches for all versions of a package and the other containing only patches for the current version of the package, you could use set.difference() to find the items that are available to be applied that won't be applied and then emit the warnings.

Reply:
How can I build a set for patches for current version? Should I add a method in overrides doing something like:

```
def patches_directory_for_current_version(patches_dir: pathlib.Path, source_dir_name: str) -> pathlib.Path | None:
    """
    This method will return patches directory for current version
    """
    return patches_dir / source_dir_name if (patches_dir / source_dir_name).exists() else None
```
Or is there  a better way? I am not sure